CadRevealRvmProvider/Converters/ConverterExceptionHandling.cs

```diff
@@ -0,0 +1,534 @@
+﻿namespace CadRevealRvmProvider.Converters;
+
+using CadRevealComposer.Utils;
+using RvmSharp.Primitives;
+using System.Diagnostics;
+using System.Numerics;
+
+public static class ConverterExceptionHandling
+{
+    public static bool CanBeConverted(
+        this RvmBox rvmBox,
+        Vector3 scale,
+        Quaternion rotation,
+        FailedPrimitivesLogObject failedPrimitivesLogObject
+    )
+    {
+        if (
+            !(
+                float.IsFinite(rotation.X)
+                && float.IsFinite(rotation.Y)
+                && float.IsFinite(rotation.Z)
+                && float.IsFinite(rotation.W)
+            )
+        )
+        {
+            if (failedPrimitivesLogObject != null)
+            {
+                failedPrimitivesLogObject.FailedBoxes.RotationCounter++;
+                return false;
+            }
+        }
+
+        if ((rvmBox.LengthX <= 0 || rvmBox.LengthY <= 0 || rvmBox.LengthZ <= 0))
+        {
+            if (failedPrimitivesLogObject != null)
+                failedPrimitivesLogObject.FailedBoxes.SizeCounter++;
+
+            return false;
+        }
+
+        if (scale.X < 0 || scale.Y < 0 || scale.Z < 0)
+        {
+            if (failedPrimitivesLogObject != null)
+                failedPrimitivesLogObject.FailedBoxes.ScaleCounter++;
+
+            return false;
+        }
+
+        Trace.Assert(scale.IsUniform(), $"Expected Uniform Scale For Box. Was: {scale}");
+
+        if (!scale.X.ApproximatelyEquals(scale.Y, 0.0001))
+        {
+            Console.WriteLine("Warning: Found box with non-uniform X and Y scale");
+        }
```

Comment:
Are these two testing the same thing?

I think maybe the latter is from a special case mentioned in RvmCylinderConverter

We should also run a test on JSB (JS P2) to make sure that nothing breaks